apps/labs/posts/numba-polynomial-support.md

```diff
@@ -0,0 +1,64 @@
+
+Numba developers keep track of unsupported NumPy functions in [issue #4074](https://github.com/numba/numba/issues/4074). My first task was simple: pick functions from the list and implement them. I started by searching the list for functions that have already been implemented or ones that have been deprecated by NumPy. Then, I decided to pick the function `numpy.geomspace`, since a very similar function, `numpy.logspace`, had been implemented already, and this was a great starting point for me. This helped me introduce myself to the Numba development workflow and the repository structure.
+
+After my first Pull Request was merged, I moved on to other functions from the list. Other NumPy functions that I added support as a Quansight Labs intern include `np.vsplit`, `np.hsplit`, `np.dsplit`, `np.rowstack`, `np.diagflat`, `np.resize`, `np.indices`, `np.unwrap`. A great guide to using the `@overload` decorator is given in the [Numba documentation](https://numba.readthedocs.io/en/stable/extending/overloading-guide.html#a-guide-to-using-overload).
+
+## The NumPy polynomial API
+When choosing functions to implement, I was always looking for groups of similar functions. That way, I could get more functions checked from the list more quickly. This led me to implementing `polyadd, polysub, polymul` from `numpy.polynomial.polynomial` simultaneously. After seeing this, my mentor asked me if I would like to focus on support for the NumPy polynomial package. I agreed since I felt like this would be a good way of working with lower levels of Numba. In addition to adding other functions from the polynomial API, the biggest task during my internship was implementing the `Polynomial` class. It is fairly simple as it has three parameters - `coef`, `domain`, and `window`, each of which is a NumPy array. This meant that the foundation of the implementation was fairly straightforward. Unfortunately, there were a lot of details in the behaviour of the class, and each small detail seemed challenging when I had to implement it using the low-level API. The most interesting thing about the implementation of the class is that the two optional parameters - `domain` and `window`, have default values that are NumPy arrays of integer dtype, whereas if the user decides to set `domain` and `window`, then these arrays are always cast to arrays of double dtype. This meant that the user could never set the values of these two parameters to their default values.
```

Comment:
```suggestion
When choosing functions to implement, I was always looking for groups of similar functions. That way, I could get more functions checked from the list more quickly. This led me to implementing `polyadd`, `polysub` and `polymul` from `numpy.polynomial.polynomial` simultaneously. After seeing this, my mentor asked me if I would like to focus on support for the NumPy polynomial package. I agreed since I felt like this would be a good way of working with lower levels of Numba. In addition to adding other functions from the polynomial API, the biggest task during my internship was implementing the [`Polynomial`](https://numpy.org/doc/stable/reference/generated/numpy.polynomial.polynomial.Polynomial.html) class. It is fairly simple as it has three parameters - `coef`, `domain`, and `window`, each of which is a NumPy array. This meant that the foundation of the implementation was fairly straightforward. Unfortunately, there were a lot of details in the behaviour of the class, and each small detail seemed challenging when I had to implement it using the low-level API. The most interesting thing about the implementation of the class is that the two optional parameters - `domain` and `window`, have default values that are NumPy arrays of integer dtype, whereas if the user decides to set `domain` and `window`, then these arrays are always cast to arrays of double dtype. This meant that the user could never set the values of these two parameters to their default values.
```

Reply:
You can also mention the NumPy pull request (https://github.com/numpy/numpy/pull/24745)